sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -209,26 +216,148 @@ private HttpResponse receiveResponse(HttpRequest httpRequest, HttpURLConnection
                 }
             }
 
-            AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
-
-            try (InputStream errorStream = connection.getErrorStream();
-                 InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
-                byte[] buffer = new byte[8192];
-                int length;
-                while ((length = inputStream.read(buffer)) != -1) {
-                    outputStream.write(buffer, 0, length);
+            if (!eagerlyReadResponse) {
+                if (responseHeaders.get(HeaderName.CONTENT_TYPE).getValue().contains(TEXT_EVENT_STREAM) && listener != null) {
+                    try {
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                        processBuffer(reader);
+                    } catch (IOException e) {
+                        retryExceptionForSSE(e);
+                    }
+                }
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders);
+            } else {
+                AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
+
+                try (InputStream errorStream = connection.getErrorStream();
+                     InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
+                    byte[] buffer = new byte[8192];
+                    int length;
+                    while ((length = inputStream.read(buffer)) != -1) {
+                        outputStream.write(buffer, 0, length);
+                    }
                 }
-            }
 
-            return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
-                BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
+                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+            }
         } catch (IOException e) {
             throw LOGGER.logThrowableAsError(new RuntimeException(e));
+
         } finally {
             connection.disconnect();
         }
     }
 
+    /**
+     * Processes the sse buffer and dispatches the event
+     * @param reader The BufferedReader object
+     * @throws IOException If an I/O error occurs
+     */
+    private void processBuffer(BufferedReader reader) throws IOException {
```

Comment:
Updated the DefaultHttpClient to be stateless again. 
In case of exceptions, we do need the `retryAfter` and `lastEventID` properties which are now returned from this method.